File: samples/java/eventhub/src/main/java/com/contoso/kafka/KafkaOutputManyWithHeaders.java
package com.contoso.kafka;

import com.microsoft.azure.functions.annotation.*;
import com.microsoft.azure.functions.*;

import java.util.Optional;

public class KafkaOutputManyWithHeaders {
    @FunctionName("KafkaOutputManyWithHeaders")
    public HttpResponseMessage run(
            @HttpTrigger(name = "req", methods = {HttpMethod.GET, HttpMethod.POST}, authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage<Optional<String>> request,
            @KafkaOutput(
                name = "kafkaOutput",
                topic = "topic",  
                brokerList="%BrokerList%",
                username = "$ConnectionString", 
                password = "EventHubConnectionString",
                authenticationMode = BrokerAuthenticationMode.PLAIN,
                // sslCaLocation = "confluent_cloud_cacert.pem", // Enable this line for windows.  
                protocol = BrokerProtocol.SASLSSL
            )  OutputBinding<KafkaEntity[]> output,
            final ExecutionContext context) {
        context.getLogger().info("Java HTTP trigger processed a request.");
        KafkaEntity[] kevents = new KafkaEntity[2];
        KafkaHeaders[] headersForEvent1 = new KafkaHeaders[1];
        headersForEvent1[0] = new KafkaHeaders("test", "java");
        KafkaEntity kevent1 = new KafkaEntity(364, 0, "topic", "2022-04-09T03:20:06.591Z", "one", headersForEvent1);

        KafkaHeaders[] headersForEvent2 = new KafkaHeaders[1];
        headersForEvent2[0] = new KafkaHeaders("test1", "java");
        KafkaEntity kevent2 = new KafkaEntity(364, 0, "topic", "2022-04-09T03:20:06.591Z", "two", headersForEvent2);
        
        kevents[0] = kevent1;
        kevents[1] = kevent2;
        output.setValue(kevents);
        return request.createResponseBuilder(HttpStatus.OK).body("Ok").build();
    }
}

File: samples/dotnet/Confluent/KafkaOutputManyWithHeaders.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.Azure.WebJobs.Extensions.Kafka;
using Microsoft.Extensions.Logging;

namespace Confluent
{
    public class KafkaOutputManyWithHeaders
    {
        [FunctionName("KafkaOutputManyWithHeaders")]
        public static IActionResult Output(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest req,
            [Kafka("BrokerList",
                    "topic",
                    Username = "ConfluentCloudUserName",
                    Password = "ConfluentCloudPassword",
                    Protocol = BrokerProtocol.SaslSsl,
                   AuthenticationMode = BrokerAuthenticationMode.Plain
            )] out KafkaEventData<string>[] eventDataArr,
            ILogger log)
        {
            log.LogInformation("C# HTTP trigger function processed a request.");
            eventDataArr = new KafkaEventData<string>[2];
            eventDataArr[0] = new KafkaEventData<string>("one");
            eventDataArr[0].Headers.Add("test", System.Text.Encoding.UTF8.GetBytes("dotnet"));
            eventDataArr[1] = new KafkaEventData<string>("two");
            eventDataArr[1].Headers.Add("test1", System.Text.Encoding.UTF8.GetBytes("dotnet"));
            return new OkObjectResult("Ok");
        }
    }
}